### DISCUSSION TITLE
How to reduce the message from health check.

### DISCUSSION THREAD

**Discussion by shizunge** [firefly-iii/firefly-iii#7099] [Firefly III]
I am using docker, and saw the message from health check.

I want to keep the health check, but I really don't need the messages. 

How not to generate the healcheck messages?

**Comment by JC5** [comment]
Why?

**Reply by diamant-x** [reply]
Apologies for necroposting, but I just came across this.
The lines "127.0.0.1 - - [09/May/2024:10:29:21 +0200] "GET /health HTTP/1.1" 200 1741 "-" "curl/7.74.0"" push any other important logs away and at some point pruned from the logger of docker.
If I would want to troubleshoot a daily or monthly cron for handling the auto-add of monthly budgets, or something like that, then it is buried and, because those processes normally run overnight, probably already prunned away from the docker logs.

In general is more efficient and [useful ](https://prince2.wiki/management-products/exception-report/)to log by exception.

**Reply by JC5** [reply]
Feel free to suggest the necessary Apache configuration.

**Reply by diamant-x** [reply]
Not a PHP developer, but could it be those logs get triggered by the database health-check call which seems to trigger the following? https://github.com/firefly-iii/firefly-iii/blob/5c2397bbae4557592c57c4bac6c57bbee48060b3/app/Http/Controllers/System/HealthcheckController.php#L42
Maybe such return could return a console.log, so it only prints if the global app log level is set, or, otherwise a 'checkException' method so that return is null unless the check fails?

Apologies if such line is not coming from there.

**Comment by shizunge** [comment]
Why do we need periodic messages?

I am using docker. The docker already tracks whether the service is healthy
or not.

I am trying increase the app log level from notice to warning. It does not
help disabling the health checking messages.

**Reply by JC5** [reply]
Sorry, the logs are generated by Apache. I can't help you I'm afraid.

**Reply by shizunge** [reply]
Where can I find the Dockerfile?

I think we can add some conditions to log in Apache https://httpd.apache.org/docs/2.4/logs.html to filter out the healthy check messages.

**Reply by JC5** [reply]
Who's the "we" here? I'm fine with these messages. It's one line every minute.

**Reply by shizunge** [reply]
I want to play with the Apache config firstly. But I don't know where to start.
Reading the Dockerfile, seeing how things are built, may be a good start point, yet I cannot find it.

**Reply by JC5** [reply]
Check out https://github.com/firefly-iii/docker 👍
